lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -532,133 +550,165 @@ class PullbackInfo {
+      std::string structName;
+      switch (kind) {
+      case AutoDiffAssociatedFunctionKind::JVP:
+        structName = "Differential";
+        break;
+      case AutoDiffAssociatedFunctionKind::VJP:
+        structName = "Pullback";
+        break;
+      }
+      s << structName << " struct created for function @" << original->getName()
         << " bb" << originalBB->getDebugID() << '\n';
-      pullbackStruct->print(s);
+      linearMapStruct->print(s);
       s << '\n';
     });
-    return pullbackStruct;
+    return linearMapStruct;
   }
 
 public:
-  PullbackInfo(const PullbackInfo &) = delete;
-  PullbackInfo &operator=(const PullbackInfo &) = delete;
+  LinearMapInfo(const LinearMapInfo &) = delete;
+  LinearMapInfo &operator=(const LinearMapInfo &) = delete;
 
-  explicit PullbackInfo(ADContext &context, SILFunction *original,
-                        SILFunction *vjp, const SILAutoDiffIndices &indices);
+  explicit LinearMapInfo(
+      ADContext &context, AutoDiffAssociatedFunctionKind kind,
+      SILFunction *original, SILFunction *assocFn,
+      const SILAutoDiffIndices &indices);
 
-  /// Returns the pullback struct associated with the given original block.
-  StructDecl *getPullbackStruct(SILBasicBlock *origBB) const {
-    return pullbackStructs.lookup(origBB);
+  /// Returns the linear map struct associated with the given original block.
+  StructDecl *getLinearMapStruct(SILBasicBlock *origBB) const {
+    return linearMapStructs.lookup(origBB);
   }
 
-  /// Returns the lowered SIL type of the pullback struct associated with the
+  /// Returns the lowered SIL type of the linear map struct associated with the
   /// given original block.
-  SILType getPullbackStructLoweredType(SILBasicBlock *origBB) const {
-    auto *pbStruct = getPullbackStruct(origBB);
+  SILType getLinearMapStructLoweredType(SILBasicBlock *origBB) const {
+    auto *pbStruct = getLinearMapStruct(origBB);
     auto pbStructType =
         pbStruct->getDeclaredInterfaceType()->getCanonicalType();
     return typeConverter.getLoweredType(pbStructType,
                                         ResilienceExpansion::Minimal);
   }
 
-  /// Returns the predecessor enum associated with the given original block.
-  EnumDecl *getPredecessorEnum(SILBasicBlock *origBB) const {
-    return predecessorEnums.lookup(origBB);
+  /// Returns the linear map enum associated with the given original block.
+  EnumDecl *getLinearMapEnum(SILBasicBlock *origBB) const {
+    return linearMapEnums.lookup(origBB);
   }
 
-  /// Returns the lowered SIL type of the predecessor enum associated with the
+  /// Returns the lowered SIL type of the linear map enum associated with the
   /// given original block.
-  SILType getPredecessorEnumLoweredType(SILBasicBlock *origBB) const {
-    auto *predEnum = getPredecessorEnum(origBB);
+  SILType getLinearMapEnumLoweredType(SILBasicBlock *origBB) const {
+    auto *predEnum = getLinearMapEnum(origBB);
```

Comment:
```suggestion
    auto *linMapEnum = getLinearMapEnum(origBB);
```

Please generalize all references to enums to "linMapEnum".